roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,76 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+from openshift_checks.mixins import NotContainerizedMixin
+
+
+class DockerImageAvailability(NotContainerizedMixin, OpenShiftCheck):
+    """Check that required Docker images are available."""
+
+    name = "docker_image_availability"
+    tags = ["preflight"]
+
+    # attempt to pull required docker images and fail if
+    # any images are missing, or error occurs during pull
+    def run(self, tmp, task_vars):
+        openshift_release = get_var(task_vars, "openshift_release")
+        openshift_image_tag = get_var(task_vars, "openshift_image_tag")
+
+        changed = False
+        failed_pulls = set()
+
+        pulled_images = self.pull_images(self.add_image_tags(self.rpm_docker_images(), openshift_release), tmp, task_vars)
+        if "failed_images" in pulled_images:
+            failed_pulls.update(pulled_images["failed_images"])
+
+        if not openshift_image_tag:
+            if len(failed_pulls) < len(self.rpm_docker_images()) + len(self.rpm_qualified_docker_images()):
+                changed = True
+            return {"changed": changed, "failed": True, "msg": "Unable to retrieve fully qualified image tag. Failed to fetch images: %s" % (set(failed_pulls).update(self.rpm_qualified_docker_images()))}
+
+        # pull each required image that requires a qualified image
+        # tag (v0.0.0.0) rather than a standard release format tag (v0.0)
+        pulled_qualified_images = self.pull_images(self.add_image_tags(self.rpm_qualified_docker_images(), "v" + openshift_image_tag), tmp, task_vars)
```

Comment:
We don't currently install skopeo as part of the install, but as we move towards system containers, it will become a hard dependency at some point. Even without skopeo, it may make sense to default to querying the registry API to search tags for images, rather than pulling images. It would also allow us to be a bit fuzzier on finding a matching container image.